pokecat/__init__.py

```diff
@@ -217,15 +217,20 @@ def populate_pokeset(pokeset, skip_ev_check=False):
 
     # check gender
     gender = pokeset["gender"]
+    expected_gender = pokeset["species"]["gender"]
+    if gender is None:
+        gender = expected_gender
+    if gender == "mf":
+        gender = ["m", "f"]
     if not isinstance(gender, list):
         gender = [gender]
     if not gender:
         raise ValueError("List of possible genders cannot be empty.")
     for gender_single in gender:
         if gender_single not in ("m", "f", None):
             raise ValueError("gender can only be 'm', 'f' or not set (null), but not %s" % (gender_single,))
-    if len(gender) > 1 and None in gender:
-        raise ValueError("non-gender cannot be mixed with m/f")
+        if not ((expected_gender and gender_single) and (gender_single in expected_gender) or gender_single == expected_gender):
+            raise ValueError(f"Incorrect gender for {pokeset['species']['name']}")
```

Comment:
Should it be impossible to set a gendered pokemon's gender to null, or give an otherwise genderless pokemon a gender? If there are no use cases for this, sure. Otherwise this could be a warning

Reply:
Also this condition is confusing. Shouldn't `expected_gender` be change to be a list as well? And can you add a test to the unittest-file for this new case?

Reply:
Once again, setting a gendered pokemon's gender to null is already discouraged, and scripts that analyze pokesets already raise an error when this happens (assuming there are already gendered sets of the same species, which is basically a given). It can be a warning if you prefer, however.

As for the format of the gender in the species, if you would prefer it as a list, then I will change that. This will indeed make that condition more readable

What unittest file? I can't find one in this repo. However, I have successfully tested this with test cases similar to the examples I listed above.

Reply:
pokecat_test.py

Reply:
Under the given circumstances I'd say keep it an error. There's no usecase for changing the gendering anyway